tests/test_polymorphic_model.py

```diff
@@ -0,0 +1,132 @@
+from typing import Optional
+
+from sqlalchemy import ForeignKey
+from sqlalchemy.orm import mapped_column
+from sqlmodel import Field, Session, SQLModel, create_engine, select
+
+from tests.conftest import needs_pydanticv2
+
+
+@needs_pydanticv2
+def test_polymorphic_joined_table(clear_sqlmodel) -> None:
+    class Hero(SQLModel, table=True):
```

Comment:
Would love to see a syntax that's more focused on what the user wants and hides implementation details.
Example:

```
from fquery.sqlmodel import sqlmodel

@sqlmodel
class Hero:
  hero_type: str = polymorphic(default="hero")
  
@sqlmodel
class DarkHero(Hero):
   dark_power: str = field(default="dark", metadata=...)
```   

You can then perform metaprogramming magic inside the `sqlmodel` decorator to compute exactly the same class definitions you have in the PR here.

Reply:
Polymorphic orm have a lot of details, and these details are not concerned at dataclass level. So I think keep the origin sqlalchemy syntax is the best choice for early version. Maybe we can improve it at next PR.

Reply:
I think the key phrase in your comment is **for early version**. If `sqlalchemy `syntax was good enough, `sqlmodel `wouldn't exist :)

How other ORMs are doing it:

* [Ruby on rails](https://guides.rubyonrails.org/association_basics.html#polymorphic-associations)
* [zenstack/prisma](https://zenstack.dev/blog/polymorphism)

I think the key points are:

* Decorator approach hides inheritance for those who don't like it. Instead it injects behavior.
* `fquery.sqlmodel` already has `one_to_many()`. Perhaps it could be enhanced with `one_to_many("hero_type")`

But like you say - probably best done in two steps.